Grow.c

```diff
@@ -3566,33 +3597,24 @@ static int reshape_array(char *container, int fd, char *devname,
 		return 1;
 	}
 
-	if (!forked)
-		if (continue_via_systemd(container ?: sra->sys_name,
-					 GROW_SERVICE, NULL)) {
-			free(fdlist);
-			free(offsets);
-			sysfs_free(sra);
-			return 0;
-		}
-
 	/* Now we just need to kick off the reshape and watch, while
 	 * handling backups of the data...
 	 * This is all done by a forked background process.
 	 */
-	switch(forked ? 0 : fork()) {
+	switch (handle_forking(forked, container ?: sra->sys_name)) {
```

Comment:
Please check if there is something that can be passed here for native (if that is a case).

Reply:
if not, probably some assert that for native it must be forked, would be helpful. I'm not sure here.

Reply:
Looks like rn we rely on service start to fail for native, and default to forking. Cursed.
![image](https://github.com/user-attachments/assets/da4b201b-412c-4e2f-bd29-175947f2025c)

Reply:
I can't think of scenario for native in which code reaches here, nor scenario for imsm in which sys_name will be used. 
Note that `container ?: sra->sys_name` will pass `container` if it's not null.

@mtkaczyk I would opt for leaving this as it is. This was "working" all these years when "--freeze-reshape" was a thing.
This already is quite dangerous change, can we deal with all "container" thing later?

Reply:
Discussed offline